articles/151_node_to_participant_mapping.md

```diff
@@ -0,0 +1,220 @@
+The `Participant` GUID can be used as the key.
+This helps for keeping only one message per `Participant` in the history (see `QoS for communicating node information`).
+The rest of the message will be a sequence of with the information of each node.
+That message should contain the `Node` name, and four sequences:
+- GUID of its `Publishers`
+- GUID of its `Subscriptions`
+- GUID of its `Services`
+- GUID of its `Clients`
+
+Vector bounds: TBD
+
+##### QoS for communicating node information
+
+Each published message should be available to late `Subscribers`, and only the last message of each key should be kept.
+For that reason, the QoS of the `Publishers` should be:
+
+- Durability: Transient Local
+- History: Keep Last
+- History depth: 1
+- Reliability: Reliable
+
+Considering that a keyed topic will be used, in which the history depth apply for each key, only one `Publisher` per process will be needed.
+In that case, `unregister_instance` can be used for delete that key from the history (see [RTI Managing Data Instances](https://community.rti.com/static/documentation/connext-dds/5.2.3/doc/manuals/connext_dds/html_files/RTI_ConnextDDS_CoreLibraries_UsersManual/Content/UsersManual/Managing_Data_Instances__Working_with_Ke.htm)).
+
+
+The configuration of the `Subscriber` QoS depends on how the data will be accessed later:
+- Polled using `Subscriber` read method when needed.
+- Listened and organized in a local cache.
+
+The second option allows better organization of this information (e.g.: in hash tables).
+In the first case, the QoS of the `Subscriber` should be:
+
+- Durability: Transient Local
+- History: Keep Last
+- History depth: 1
+- Reliability: Reliable
+
+In the second case, durability can be changed to `Volatile`.
+
+#### Using USER_DATA and GROUP_DATA QoSPolicy
+
+Each `Participant` could have in its user data the list of node names that owns.
+When this data is changed, each `ParticipantListener` will be notified.
+This is not a good option, as `UserData` is just a sequence of bytes.
+Organizing a complex message in it won't be easy nor performant.
+
+Similarly to `UserData`, `GroupData` is a available in `Publishers` and `Subscribers`.
+These entities only need to communicate the GUID of the `Participant` and the `Node` name from which it was created.
+This idea can be combined with a topic just publishing the list of `Node` names, without including all the other vectors in the message.
+Although, it is more difficult to communicate this information for `Services` and `Clients`, as they use behind the scenes just a `DDS Publisher` and `Subscriber`.
+
+### Other implications
+
+#### Security
+
+In `DDS`, security can be specified at a `Participant` level.
+If one `Node` is mapped to one `Participant`, individual configuration of its security key and access control policy is possible.
+From a security point of view, only being able to configure it at a `Participant` (or per process) level is enough.
```

Comment:
I think that composing nodes that are not trusted in the same process is not secure, regardless of control access policies, because they share the same address space.
If you want to run a  `Node` you don't trust, you should run it in another process.
In that case, configuring different access control policies will be possible.